tests/components/application-connector/internal/testkit/test-api/apis.go

```diff
@@ -11,13 +11,20 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-func SetupRoutes(logOut io.Writer, basicAuthCredentials BasicAuthCredentials) http.Handler {
+func SetupRoutes(logOut io.Writer, basicAuthCredentials BasicAuthCredentials, oAuthCredentials OAuthCredentials) http.Handler {
 	router := mux.NewRouter()
 
 	router.HandleFunc("/v1/health", alwaysOk).Methods("GET")
 	api := router.PathPrefix("/v1/api").Subrouter()
 	api.Use(Logger(logOut, logger.DevLoggerType))
 
+	oauth := NewOAuth(oAuthCredentials.ClientID, oAuthCredentials.ClientSecret)
+
+	{
+		api.HandleFunc("/oauth/token", oauth.Token).Methods(http.MethodPost)
```

Comment:
there is already an `/oauth` block, starting from line 40

Reply:
I wanted to separate the token endpoints and the _target url endpoints_ as the token ones will be used across also the CSRF test but the _target urls_ will be different. In this case, there will be shared token urls and separated _target urls_ for each test case

Reply:
We could consider naming endpoint blocks in the future 🤔